process/sync/storageBootstrap/shardStorageBootstrapper.go

```diff
@@ -183,6 +187,11 @@ func (ssb *shardStorageBootstrapper) cleanupNotarizedStorageForHigherNoncesIfExi
 				"nonce", metaBlock.GetNonce(),
 				"hash", metaBlockHash,
 				"error", err.Error())
+		} else {
+			log.Debug("meta block has been removed from MetaHdrNonceHashDataUnit storage",
```

Comment:
In case we need to rollback at the end of the epoch (can happen) we will get max 28800 prints. I would remove L191-L194 + L204-L2018

Reply:
This method will be called and will remove only nonces higher than last cross notarized header if they exist, no matter if the node will roll back 28800 nonces from the last cross notarized header